components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/config/ServiceComponent.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.siddhi.apps.api.rest.config;
+
+import org.osgi.service.component.annotations.*;
+import org.wso2.carbon.analytics.idp.client.core.api.AnalyticsHttpClientBuilderService;
+import org.wso2.carbon.config.provider.ConfigProvider;
+
+@Component(
+        name = "org.wso2.carbon.siddhi.apps.api.rest.config.ServiceComponent",
+        service = ServiceComponent.class,
+        immediate = true
+)
+public class ServiceComponent {
+
+    @Activate
+    public void start(){
+
```

Comment:
Shouldn't there be permission validated?

Reply:
Implemented permission provider in 870c965